lib/static/new-ui/features/suites/components/SuitesTreeView/utils.ts

```diff
@@ -286,21 +313,36 @@ export const sortTreeNodes = (entitiesContext: EntitiesContext, treeNodes: TreeN
                     const runsCount = groupId ? browser.resultIds.filter(resultId => groups[groupId].resultIds.includes(resultId)).length : browser.resultIds.length;
 
                     return createWeight([1, runsCount], {runsCount});
+                } else if (currentSortExpression.type === SortType.ByDuration) {
+                    const browser = browsers[treeNode.data.entityId];
+                    const groupId = getGroupId(treeNode.data);
+                    const resultIds = groupId ? browser.resultIds.filter(resultId => groups[groupId].resultIds.includes(resultId)) : browser.resultIds;
+                    const totalTime = resultIds.reduce((accTime, resultId) => accTime + (results[resultId].duration ?? 0), 0);
+
+                    return createWeight([totalTime, treeNode.data.title.join(' ')], {runsCount: resultIds.length, duration: totalTime});
+                } else if (currentSortExpression.type === SortType.ByStartTime) {
+                    const browser = browsers[treeNode.data.entityId];
+                    const startTime = results[browser.resultIds[0]].timestamp;
+
+                    return createWeight([startTime], {startTime});
                 }
                 break;
             }
         }
 
-        notifyOfUnsuccessfulWeightComputation();
-        return createWeight([0]);
+        return createInvalidWeight(treeNode);
     };
 
     const aggregateWeights = (weights: TreeNodeWeight[]): TreeNodeWeight => {
         if (!currentSortExpression || currentSortExpression.type === SortType.ByName) {
-            return createWeight([0]);
+            return createInvalidWeight();
         }
 
-        if (currentSortExpression.type === SortType.ByFailedRuns || currentSortExpression.type === SortType.ByTestsCount) {
+        if (
+            currentSortExpression.type === SortType.ByFailedRuns ||
+            currentSortExpression.type === SortType.ByTestsCount ||
+            currentSortExpression.type === SortType.ByDuration
+        ) {
```

Comment:
Why not `[SortType.ByFailedRuns, SortType.ByTestsCount, SortType.ByDuration].includes(currentSortExpression.type)`?